urlfcc351a1ba_kuntapavankumar_Z_tgz-pJ8-edu_concordia_dpis_ReplicaTestJ8/src/edu/concordia/dpis/MulticastServer.java

```diff
@@ -0,0 +1,86 @@
+package edu.concordia.dpis;
+
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.SocketException;
+
+@Deprecated
+public class MulticastServer extends Thread {
+	
+	private long FIVE_SECONDS = 5000;
+	private int itHasMoreRequest = 1;
+	private int port;
+	
+	
+	public MulticastServer(int port){
+		this.port = port;	
+	}
+
+	public void run(){
+		
+		do
+		{
+			DatagramSocket socket = null;
+			try
+			{	System.out.println("server is up");
+				socket = new DatagramSocket();
+				byte[] buffer = new byte[1024];
+				buffer = "just the testing code".getBytes();
+				InetAddress group = InetAddress.getByName("230.0.0.1");
+				DatagramPacket request = new DatagramPacket(buffer, buffer.length,group,port);
+				socket.send(request);
+				/*
+				//receiving the reply
+				byte[] replybuffer = new byte[1024];
+				DatagramPacket reply = new DatagramPacket(replybuffer, replybuffer.length);
+				socket.receive(reply);
+				System.out.println(new String(reply.getData()));
+				*/
+				
+			
+			} catch (SocketException e) {
+			System.out.println("SOCKET:" + e.getMessage());
+			} catch (Exception e) {
+			System.out.println("IO:" + e.getMessage());
+			}/*finally {
+				if (null != socket)
+					socket.close();
+			}*/
+		
+			getReply(socket); 
+			// sleep for a while for the request processing at other replicas
+			try
+			{
+				sleep((long)(Math.random() * FIVE_SECONDS));
+			} 
+			catch (InterruptedException e) 
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)